src/components/SearchBar.js

```diff
@@ -133,6 +178,86 @@ class SearchBar extends React.Component<SearchBarDefaultProps, SearchBarProps, S
     );
   }
 
+  // queries the index for the saved searches and sets the response to state
+  getSavedSearches() {
+    const username = AuthUtils.getLoggedInUserId();
+    const searcher = this.context.searcher;
+    const { savedSearchWorkflow, savedSearchTable, savedSearchZone } = this.props;
+    if (searcher) {
+      const qr = new SimpleQueryRequest();
+      qr.query = `AND(table:${savedSearchTable}, username_s:"${username}")`;
+      qr.facets = [];
+      qr.restParams.set('zones', [`${savedSearchZone}`]);
+      if (savedSearchWorkflow) {
+        qr.workflow = savedSearchWorkflow;
+      }
+      qr.queryLanguage = 'advanced';
+      searcher.doCustomSearch(qr, (response: QueryResponse | null, error: string | null) => {
+        if (response) {
+          this.setState(
+            {
+              response,
+            },
+            this.populateSavedSearches,
+          );
+        } else if (error) {
+          this.setState({ error });
+        }
+      });
+    }
+  }
+
+  // applies the saved search which includes a query and facet filters
+  applySavedSearch(query: string) {
+    const { location } = this.props;
+    this.props.history.push({
+      pathname: location.pathname,
+      search: query,
+    });
+  }
+
+  // populates the saved searches in the saved searches drop down
+  populateSavedSearches() {
+    const response = this.state.response;
+    if (response) {
+      const menuItemList = [];
+      response.documents.forEach((doc) => {
+        const savedSearch = {};
+        const htmlString = doc.getFirstValue('title');
+        savedSearch.id = doc.getFirstValue('.id');
+        savedSearch.title = htmlString.replace(/<[^>]+>/g, '');
+        savedSearch.query = doc.getFirstValue('query_s');
+        savedSearch.queryString = doc.getFirstValue('query_string_s');
+        let savedSearchTitle = savedSearch.title;
+        savedSearchTitle = savedSearchTitle.length > 15 ? `${savedSearchTitle.substring(0, 15)}...` : savedSearchTitle;
+        menuItemList.push(
+          <MenuItem key={savedSearch.id}>
+            <span // eslint-disable-line
+              onClick={() => {
+                this.applySavedSearch(savedSearch.query);
+              }}
+              title={savedSearch.title}
+            >
+              {savedSearchTitle}
+            </span>
+            <span // eslint-disable-line
+              onClick={() => {
+                this.deleteThisSearch(savedSearch.id);
+              }}
+              style={{
+                float: 'right',
+                fontWeight: 'bold',
+              }}
+            >
+              &times;
+            </span>
+          </MenuItem>,
+        );
+      });
+      this.setState({ savedSearchList: menuItemList });
```

Comment:
Storing jsx as state is generally considered bad practice. It's usually an indicator that you need to refactor your component, break out a function, or create a building block component.
https://reactjs.org/docs/thinking-in-react.html#step-3-identify-the-minimal-but-complete-representation-of-ui-state